tests/system/tests/test_groupmems.py

```diff
@@ -0,0 +1,44 @@
+"""
+Test groupmems
+"""
+
+from __future__ import annotations
+
+import pytest
+
+from framework.roles.shadow import Shadow
+from framework.topology import KnownTopology
+
+
+@pytest.mark.topology(KnownTopology.Shadow)
+def test_groupmems__add_user_as_root(shadow: Shadow):
+    """
+    :title: Add user to group as root user
+    :setup:
+        1. Create a test user
```

Comment:
I prefer to generalize the steps, making it easier to understand what the test is doing at a higher level. Something like, 

```
:setup:
1. Create test user and group
:steps:
1. Add user to group using groupmem as root
2. Check group and gshadow entry
:expectedresults:
1. User is added to group
2. Group values, memberships, and gshadow entry values are correct
```

Not much an issue here, but with test_newusers__create_users_from_stdin, those are a lot of steps. Consider it.

Reply:
mmm I like this idea. I haven't used the exact wording that you mentioned, but it's almost the same